src/DeviceInterface/ExtensionInterface.js

```diff
@@ -175,13 +201,7 @@ class ExtensionInterface extends InterfacePrototype {
 
             switch (message.type) {
             case 'ddgUserReady':
-                this.setupAutofill().then(() => {
-                    this.refreshSettings().then(() => {
-                        this.setupSettingsPage({shouldLog: true}).then(() => {
-                            return this.postInit()
-                        })
-                    })
-                })
+                this.resetAutofillUI(() => this.setupSettingsPage({shouldLog: true}))
```

Comment:
We only want to set up the settings page on `ddgUserReady` event, not every time we reset the autofill UI, so pass it in here as a callback to be called before `postInit`

Reply:
I wonder why were doing `this.setupAutofill()…` before. The UX doesn't seem to have changed, but the lack of automated tests around this makes me nervous. Do you know if removing all that stuff is safe?

Reply:
@GioSensation Nothing's been removed, these calls have just bee moved into `this.resetAutofillUI` method -- it calls `this.refreshSettings`, `this.setupAutofill`, the `callback` (which calls `this.setupSettingsPage`) and then finally `this.postInit`